src/llm/apis/openai_completions.cpp

```diff
@@ -924,7 +955,7 @@ std::string OpenAIChatCompletionsHandler::serializeUnaryResponse(const std::vect
     return buffer.GetString();
 }
 
-std::string OpenAIChatCompletionsHandler::serializeUnaryResponse(const ov::genai::EncodedResults& results) {  // TODO separate common part with function implemented above
+std::string OpenAIChatCompletionsHandler::serializeUnaryResponse(const ov::genai::EncodedResults& results, const std::string responseParserName) {  // TODO separate common part with function implemented above
```

Comment:
why do we have 2 functions, that do the same thing? I dont think is clean approach, I see that the arguments differ, but as you can see this is not maintainable

maybe we could convert one arg to another before passing to single function that does the work? or some intermediate format, not sure what would fit here, probably std::string or vector of strings (instead of EncodedResults)

(comment mentioned in group chat)